dix/dixfonts.c

```diff
@@ -764,18 +762,12 @@ doListFontsAndAliases(ClientPtr client, LFclosurePtr c)
             bufptr += names->length[i];
         }
     }
-    nnames = rep.nFonts;
-    rep.length = bytes_to_int32(stringLens + nnames);
```

Comment:
The old code is a bit tricky here. They allocate potentially more memory than needed, but at the end write only what’s needed. We need to adjust the new code to do the equivalent thing.